File: src/test/java/fi/eonwe/wikilinks/leanpages/OrderedPage.java
package fi.eonwe.wikilinks.leanpages;

import java.util.function.IntConsumer;

/**
 * Old page-format that is only used in tests anymore.
 */
public class OrderedPage {

    private final BufferWikiPage page;
    private final int[] targetIndices;

    protected OrderedPage(BufferWikiPage page, int[] targetIndices) {
        this.page = page;
        this.targetIndices = targetIndices;
    }

    public int getId() { return page.getId(); }
    public BufferWikiPage getPage() { return page; }

    public void forEachLinkIndex(IntConsumer c) {
        for (int i : targetIndices) c.accept(i);
    }

}

File: src/test/kotlin/fi/eonwe/wikilinks/leanpages/OrderedPage.kt
package fi.eonwe.wikilinks.leanpages

import java.util.function.IntConsumer

/**
 * Old page-format that is only used in tests anymore.
 */
class OrderedPage(page: BufferWikiPage, private val targetIndices: IntArray) {
    val id: Int = page.getId()

    fun forEachLinkIndex(c: IntConsumer) {
        for (i in targetIndices) c.accept(i)
    }
}